### ISSUE TITLE
VideoTexture: Doesn't handle file rotation.

### ISSUE THREAD

**Issue by MarcRibguth** [mrdoob/three.js#13671]
Hallo,
It seems that the video from some mobile devices are not rotating correctly. I used the same video element for the right view and the left plain and the rotation is not the same. Other videos are working.

![bildschirmfoto vom 2018-03-23 11-10-49](https://user-images.githubusercontent.com/26739421/37824763-fb7336dc-2e8d-11e8-92d1-139c80b46177.png)

[video.zip](https://github.com/mrdoob/three.js/files/1841144/video.zip)

##### Three.js version

- [x] r91
- [x] r89

##### Browser

- [ ] All of them
- [x] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] Linux

**Comment by MarcRibguth** [comment]
![bildschirmfoto vom 2018-03-23 11-10-49](https://user-images.githubusercontent.com/26739421/37824716-d6a382a8-2e8d-11e8-8e3b-bbff009b0511.png)

**Comment by MarcRibguth** [comment]
I can try to do thet, but I'm not shoure when I will have the time for that.
The probelm is, that I use the loaded video data to create the geometry of the mesh on the left side. That part is working fine. But when I load the video on to the plain, the video has not the right rotation like in the video element on the right side.

```
function loadedmetadataFunk(event) {

    if (event && event.path.length > 0 && event.path[0]) {

          mesh.geometry.dispose();
          mesh.geometry = new THREE.PlaneGeometry(event.path[0].videoWidth / 50, 
          event.path[0].videoHeight / 50);

    }

    showProgressSpinnerInEditor(false);
}

let vid = document.createElement("Video")
vid.src = url;
vid.load();
vid.addEventListener('loadedmetadata', loadedmetadataFunk);

let texture = new THREE.VideoTexture(vid);
texture.generateMipmaps = false;
texture.magFilter = THREE.LinearFilter;
texture.minFilter = THREE.LinearFilter;
mesh.material = new THREE.MeshBasicMaterial({ map: texture });
```
I will try to create an example ready the in next days.

**Comment by Mugen87** [comment]
Try you video with this [live demo](https://jsfiddle.net/f2Lommf5/3698/ ) and test if the orientation is correct.

**Comment by MarcRibguth** [comment]
I checked it and the rotation is wrong. I also noticed, that Windows Media-Player is also not rotating the file correctly. In VLC-Player is the rotation right.

![test](https://user-images.githubusercontent.com/26739421/37848029-0325672a-2ed3-11e8-85e2-60ddbfa6af3e.JPG)

**Comment by Mugen87** [comment]
I don't think that's a `three.js` related issue. Or do you have a suggestion what the library should make different?

**Comment by MarcRibguth** [comment]
I would expect that the texture has the same rotation like the video element I putted in. Or do you know a way to detract the rotation?

**Comment by Mugen87** [comment]
> Or do you know a way to detract the rotation?

No. And i don't think that's the task of a 3D library. 

You should investigate why Windows Media-Player shows the same behavior.

**Comment by calrk** [comment]
The reason for this is most likely the video rotation metadata that has been saved in the video when filming on a mobile device. The mobile will record the video with whatever its default rotation is, then add in the rotation metadata for video players to use when playing back. The reason I believe is to reduce any video processing needed during recording.

Modern browsers will use this to rotate the videos in playback when using <video> tags, however not when rendering to the canvas as ThreeJS does. This rotation metadata is not available to Javascript, so can't be used by ThreeJS to rotate the video.

I think the solution is to transcode the video without the rotation metadata.

**Comment by Mugen87** [comment]
@calrk Thanks for your explanation!

**Comment by mrdoob** [comment]
@kenrussell what do you think about this? the browser applies the rotation specified in the video metadata when displaying it with `<video>` but the rotation gets lost when using it as a texture.

**Comment by MarcRibguth** [comment]
I will reopen this and maybe you could make a feature out of this.

**Comment by calrk** [comment]
Yes I agree with @Mugen87 . Given my research into the problem I don't believe the solution can be implemented in `three.js` without updates to the browser video spec.

**Comment by MarcRibguth** [comment]
Maybe the video texture could load the video as file resource and read the file rotation direct?

**Comment by mrdoob** [comment]
I think the browser is going to handle this (at least chrome).
